janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/AbstractConfiguration.java

```diff
@@ -32,48 +32,97 @@
 
     protected AbstractConfiguration(ConfigNamespace root) {
         Preconditions.checkNotNull(root);
-        Preconditions.checkArgument(!root.isUmbrella(),"Root cannot be an umbrella namespace");
+        Preconditions.checkArgument(!root.isUmbrella(), "Root cannot be an umbrella namespace");
         this.root = root;
     }
 
+    protected static Configuration restrictTo(final Configuration config, final String... fixedUmbrella) {
+        Preconditions.checkArgument(fixedUmbrella != null && fixedUmbrella.length > 0);
+        return new Configuration() {
+
+            private String[] concat(String... others) {
+                if (others == null || others.length == 0) return fixedUmbrella;
+                String[] join = new String[fixedUmbrella.length + others.length];
+                System.arraycopy(fixedUmbrella, 0, join, 0, fixedUmbrella.length);
+                System.arraycopy(others, 0, join, fixedUmbrella.length, others.length);
+                return join;
+            }
+
+            @Override
+            public boolean has(ConfigOption option, boolean includeRoot, String... umbrellaElements) {
+                if (option.getNamespace().hasUmbrella())
+                    return config.has(option, includeRoot, concat(umbrellaElements));
+                else
+                    return config.has(option, includeRoot);
+            }
+
+            @Override
+            public <O> O get(ConfigOption<O> option, boolean includeRoot, String... umbrellaElements) {
+                if (option.getNamespace().hasUmbrella())
+                    return config.get(option, includeRoot, concat(umbrellaElements));
+                else
+                    return config.get(option, includeRoot);
+            }
+
+            @Override
+            public Set<String> getContainedNamespaces(ConfigNamespace umbrella, String... umbrellaElements) {
+                return config.getContainedNamespaces(umbrella, concat(umbrellaElements));
+            }
+
+            @Override
+            public Map<String, Object> getSubset(ConfigNamespace umbrella, String... umbrellaElements) {
+                return config.getSubset(umbrella, concat(umbrellaElements));
+            }
+
+            @Override
+            public Configuration restrictTo(String... umbrellaElements) {
+                return config.restrictTo(concat(umbrellaElements));
+            }
+        };
+    }
+
```

Comment:
Why did you move this up?

Reply:
Ah! It's my IDE formatting result. There is no reason.

Reply:
Could you revert this all the format changes make this pr hard to review

Reply:
OK. I did.